force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -163,9 +185,13 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer generateRandomNumber() {
         // Use Math class to generate number between 50 and 100
-
+        Double randomNum = Integer.valueof((Math.random() * 100));
+        while (randomNum > 100 || randomNum < 50){
+            randomNum = Integer.valueof((Math.random() * 100));
+        }
+        Integer result = randomNum.intValue();
         // Return the random integer
-        return null; // Replace null with the variable you used to store the result
+        return result;
     }
```

Comment:
Can you solve this without the loop? If you set up your call to Math.random correctly it  will return a value between 50 and 100